src/core/mapping/base_mapper.cc

```diff
@@ -217,11 +217,13 @@ void BaseMapper::slice_task(const Legion::Mapping::MapperContext ctx,
   if (task.sharding_space.exists())
     sharding_domain = runtime->get_index_space_domain(ctx, task.sharding_space);
 
-  auto lo = key_functor->project_point(sharding_domain.lo(), sharding_domain);
-  auto hi = key_functor->project_point(sharding_domain.hi(), sharding_domain);
+  auto lo                 = key_functor->project_point(sharding_domain.lo(), sharding_domain);
+  auto hi                 = key_functor->project_point(sharding_domain.hi(), sharding_domain);
+  int32_t shards_per_proc = (sharding_domain.get_volume() + local_range.total_proc_count() - 1) /
+                            local_range.total_proc_count();
```

Comment:
I don't think the `key_functor` always preserves the volume of the sharding domain. it'd be safe to calculate the number of tasks using `lo` and `hi`.

Reply:
`linearize(lo, hi, hi) + 1` would do it

Reply:
I'd also call it `tasks_per_proc` instead of `shards_per_proc`.

Reply:
Thank you, changed that.